From 6d478d25de6b7550769b77edcbf8d330238542a8 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Tue, 29 Oct 2024 14:25:23 +0800
Subject: [PATCH] arm64: rockchip_defconfig: update for Linux 6.12

Enable:
CONFIG_ROCKCHIP_DW_HDMI_QP=y
CONFIG_DRM_PANTHOR=y

Remove:
CONFIG_ARM_ROCKCHIP_CPUFREQ

Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 arch/arm64/configs/rockchip_defconfig | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/configs/rockchip_defconfig b/arch/arm64/configs/rockchip_defconfig
index e6d71cb298e87e..ae949f0148f874 100644
--- a/arch/arm64/configs/rockchip_defconfig
+++ b/arch/arm64/configs/rockchip_defconfig
@@ -45,7 +45,6 @@ CONFIG_EXPERT=y
 CONFIG_KALLSYMS_ALL=y
 CONFIG_PROFILING=y
 CONFIG_ARCH_ROCKCHIP=y
-CONFIG_ARM64_ERRATUM_834220=y
 CONFIG_ARM64_ERRATUM_2441007=y
 CONFIG_ARM64_ERRATUM_1286807=y
 CONFIG_ARM64_ERRATUM_1542419=y
@@ -73,9 +72,7 @@ CONFIG_CPU_FREQ_GOV_CONSERVATIVE=y
 CONFIG_CPU_FREQ_GOV_SCHEDUTIL=y
 CONFIG_CPUFREQ_DT=y
 CONFIG_ARM_SCPI_CPUFREQ=y
-CONFIG_ARM_ROCKCHIP_CPUFREQ=y
 CONFIG_VIRTUALIZATION=y
-CONFIG_KVM=y
 CONFIG_KPROBES=y
 CONFIG_JUMP_LABEL=y
 CONFIG_MODULES=y
@@ -89,7 +86,7 @@ CONFIG_BINFMT_MISC=m
 CONFIG_ZSWAP=y
 CONFIG_ZSWAP_DEFAULT_ON=y
 CONFIG_ZSWAP_COMPRESSOR_DEFAULT_ZSTD=y
-CONFIG_ZSWAP_ZPOOL_DEFAULT_Z3FOLD=y
+CONFIG_ZSWAP_ZPOOL_DEFAULT_Z3FOLD_DEPRECATED=y
 CONFIG_ZBUD=y
 CONFIG_ZSMALLOC=y
 # CONFIG_COMPAT_BRK is not set
@@ -555,7 +552,7 @@ CONFIG_I2C_DEMUX_PINCTRL=m
 CONFIG_I2C_MUX_MLXCPLD=m
 CONFIG_I2C_CADENCE=m
 CONFIG_I2C_CBUS_GPIO=m
-CONFIG_I2C_DESIGNWARE_PLATFORM=y
+CONFIG_I2C_DESIGNWARE_CORE=y
 CONFIG_I2C_RK3X=y
 CONFIG_I2C_DIOLAN_U2C=m
 CONFIG_I2C_DLN2=m
@@ -722,6 +719,7 @@ CONFIG_ROCKCHIP_VOP2=y
 CONFIG_ROCKCHIP_ANALOGIX_DP=y
 CONFIG_ROCKCHIP_CDN_DP=y
 CONFIG_ROCKCHIP_DW_HDMI=y
+CONFIG_ROCKCHIP_DW_HDMI_QP=y
 CONFIG_ROCKCHIP_DW_MIPI_DSI=y
 CONFIG_ROCKCHIP_INNO_HDMI=y
 CONFIG_ROCKCHIP_LVDS=y
@@ -756,7 +754,7 @@ CONFIG_DRM_I2C_ADV7511=m
 CONFIG_DRM_CDNS_DSI=m
 CONFIG_DRM_DW_HDMI_CEC=m
 CONFIG_DRM_PANFROST=y
-CONFIG_DRM_PANTHOR=m
+CONFIG_DRM_PANTHOR=y
 CONFIG_DRM_POWERVR=y
 CONFIG_FB=y
 CONFIG_FB_UVESA=m
